client/apps/edit/components/content/section_container/index.jsx

```diff
@@ -0,0 +1,156 @@
+import { findIndex, findLastIndex } from 'lodash'
+import colors from '@artsy/reaction-force/dist/Assets/Colors'
+import { IconDrag } from '@artsy/reaction-force/dist/Components/Publishing'
+import { RemoveButton } from 'client/components/remove_button/index'
+
+import SectionImageCollection from '../sections/image_collection/index.coffee'
+import SectionSlideshow from '../sections/slideshow/index.coffee'
+import SectionText from '../sections/text/index.coffee'
+import { SectionEmbed } from '../sections/embed/index'
+import { SectionVideo } from '../sections/video/index'
+
+export class SectionContainer extends Component {
+  static propTypes = {
+    article: PropTypes.object.isRequired,
+    channel: PropTypes.object.isRequired,
+    editing: PropTypes.bool,
+    index: PropTypes.number.isRequired,
+    isHero: PropTypes.bool,
+    onRemoveHero: PropTypes.func,
+    onSetEditing: PropTypes.func.isRequired,
+    section: PropTypes.object.isRequired,
+    sections: PropTypes.object
+  }
+
+  onSetEditing = () => {
+    const {
+      editing,
+      index,
+      onSetEditing
+    } = this.props
+
+    const setEditing = editing ? null : index
+    onSetEditing(setEditing)
+  }
+
+  onRemoveSection = (e) => {
+    const {
+      section,
+      isHero,
+      onRemoveHero
+    } = this.props
+
+    e.stopPropagation()
+    section.destroy()
+
+    if (isHero) {
+      onRemoveHero()
+    }
+  }
+
+  getContentStartEnd = () => {
+    // TODO: move into text section
+    const { sections } = this.props
+    const types = sections && sections.map((section, i) => {
+      return { type: section.get('type'), index: i }
+    })
+    const start = findIndex(types, {type: 'text'})
+    const end = findLastIndex(types, {type: 'text'})
+
+    return { start, end }
+  }
+
+  getSectionComponent = () => {
+    const { section } = this.props
+    const sectionProps = this.getSectionProps()
+
+    switch (section.get('type')) {
+      case 'embed': {
+        return <SectionEmbed {...sectionProps} />
+      }
+      case 'image':
+      case 'image_set':
+      case 'image_collection': {
+        return <SectionImageCollection {...sectionProps} />
+      }
+      case 'text': {
+        return <SectionText {...sectionProps} />
+      }
+      case 'video': {
+        return <SectionVideo {...sectionProps} />
+      }
+      case 'slideshow': {
+        return <SectionSlideshow {...sectionProps} />
+      }
+    }
+  }
+
+  getSectionProps = () => {
+    const {
+      article,
+      channel,
+      editing,
+      index,
+      isHero,
+      onSetEditing,
+      section,
+      sections
+    } = this.props
+
+    const { end, start } = this.getContentStartEnd()
+
+    return {
+      article,
+      channel,
+      editing,
+      index,
+      isHero,
+      onSetEditing,
+      section,
+      sections,
+      setEditing: this.setEditing,
+      isContentStart: !isHero && start === index,
+      isContentEnd: !isHero && end === index
+    }
```

Comment:
Minor, but I bet there's a nice way to combine this with pulling from props. Thinking something with spread `{...sectionProps}`

Reply:
Good point, i think we might not need all of them and some not for all sections.

Reply:
And also once we connect things we probably won't have to pass this info at all :)